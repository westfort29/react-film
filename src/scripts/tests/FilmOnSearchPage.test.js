import React from 'react';
import { FilmOnSearchPage } from '../';
import renderer from 'react-test-renderer';

const FILM = {
  id: 68718,
  title: "Django Unchained",
  tagline: "Life, liberty and the pursuit of vengeance.",
  vote_average: 7.9,
  vote_count: 12144,
  release_date: "2012-12-25",
  poster_path: "https://image.tmdb.org/t/p/w500/5WJnxuw41sddupf8cwOxYftuvJG.jpg",
  overview: "With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.",
  budget: 100000000,
  revenue: 425368238,
  genres: [
    "Drama",
    "Western"
  ],
  runtime: 165
}

test('FilmOnSearchPage should contain right data', () => {
  const component = renderer.create(
    <FilmOnSearchPage
      film={ FILM }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

