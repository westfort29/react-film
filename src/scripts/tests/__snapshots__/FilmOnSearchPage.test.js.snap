// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmOnSearchPage should contain right data 1`] = `
<li
  className="films-search-result__film"
>
  <img
    alt="Django Unchained"
    className="films-search-result__film-img"
    src="https://image.tmdb.org/t/p/w500/5WJnxuw41sddupf8cwOxYftuvJG.jpg"
  />
  <span
    className="films-search-result__film-year"
  >
    2012
  </span>
  <p
    className="films-search-result__film-name"
  >
    Django Unchained
  </p>
  <p
    className="films-search-result__film-genre"
  >
    Drama&Western
  </p>
</li>
`;
